native/ripper/ripper.go

```diff
@@ -0,0 +1,341 @@
+func token(receiver Object, sourceLine int, t *Thread, args []Object) Object {
+	if len(args) != 1 {
+		return t.VM().InitErrorObject(errors.ArgumentError, sourceLine, "Expect 1 argument. got=%d", len(args))
+	}
+		
+	arg, ok := args[0].(*StringObject)
+	if !ok {
+		return t.VM().InitErrorObject(errors.TypeError, sourceLine, errors.WrongArgumentTypeFormat, classes.StringClass, arg.Class().Name)
+	}
+		
+	l := lexer.New(arg.Value().(string))
+	el := []Object{}
+	var nt gtoken.Token
+	for i := 0; ; i++ {
+		nt = l.NextToken()
+		if nt.Type == gtoken.EOF {
+			el = append(el, t.VM().InitStringObject("EOF"))
+			break
+		}
+		el = append(el, t.VM().InitStringObject(nt.Literal))
+	}
+	return t.VM().InitArrayObject(el)
+}
+
+// Internal functions ===================================================
+func init() {
+	vm.RegisterExternalClass("ripper", vm.ExternalClass("Ripper", "ripper.gb",
+		// class methods
+		map[string]vm.Method{
+			"instruction": instruction,
+			"lex":         lex,
+			"parse":       parse,
+			"token":       token,
+		},
+		// instance methods
+		map[string]vm.Method{},
+	))
+}
+
+// Other helper functions ----------------------------------------------
+
+func convertToTuple(instSet []*bytecode.InstructionSet, v *VM) *ArrayObject {
+	ary := []Object{}
+	for _, insts := range instSet {
+		hInsts := make(map[string]Object)
+		hInsts["name"] = v.InitStringObject(insts.Name())
+		hInsts["type"] = v.InitStringObject(insts.Type())
+		if insts.ArgTypes() != nil {
+			hInsts["arg_types"] = getArgNameType(insts.ArgTypes(), v)
+		}
+		ary = append(ary, v.InitHashObject(hInsts))
+		
+		aInst := []Object{}
+		for _, ins := range insts.Instructions {
+			hInst := make(map[string]Object)
+			hInst["action"] = v.InitStringObject(ins.Action)
+			hInst["line"] = v.InitIntegerObject(ins.Line())
+			hInst["source_line"] = v.InitIntegerObject(ins.SourceLine())
+			anchor, _ := ins.AnchorLine()
+			hInst["anchor"] = v.InitIntegerObject(anchor)
+			
+			aParams := []Object{}
+			for _, param := range ins.Params {
+				aParams = append(aParams, v.InitStringObject(param))
+			}
+			hInst["params"] = v.InitArrayObject(aParams)
+			
+			if ins.ArgSet != nil {
+				hInsts["arg_set"] = getArgNameType(ins.ArgSet, v)
+			}
+			
+			aInst = append(aInst, v.InitHashObject(hInst))
+		}
+		
+		hInsts["instructions"] = v.InitArrayObject(aInst)
+		ary = append(ary, v.InitHashObject(hInsts))
+	}
+	return v.InitArrayObject(ary)
+}
+
+func getArgNameType(argSet *bytecode.ArgSet, v *VM) *HashObject {
+	h := make(map[string]Object)
+	
+	aName := []Object{}
+	for _, argname := range argSet.Names() {
+		aName = append(aName, v.InitStringObject(argname))
+	}
+	h["names"] = v.InitArrayObject(aName)
+	
+	aType := []Object{}
+	for _, argtype := range argSet.Types() {
+		aType = append(aType, v.InitIntegerObject(argtype))
+	}
+	
+	h["types"] = v.InitArrayObject(aType)
+	return v.InitHashObject(h)
+}
+
+// TODO: This should finally be auto-generated from token.go
```

Comment:
I think this should be provided from token pkg as well

Reply:
I'll improve this in another PR